src/resources/affinity.js

```diff
@@ -48,15 +48,23 @@ const parsePreferredAffinity = R.map(x => {
 })
 const parseRequiredAffinity = R.compose(removeWeight, parsePreferredAffinity)
 
+const createSelfResolver = (name) => R.lift(R.when(
+  R.equals('self'), R.always({ name })
+))
+
 function createAffinities (cluster, config) {
   if (!config.hasOwnProperty('affinity')) {
     return null
   }
 
   const affinities = {}
 
-  const [podAffinities, podAntiAffinities] = splitByNegative(config.affinity.pod)
+  // Resolve `self` references before creating structures to avoid having to
+  // pass config throughout every function
+  const resolveSelfToName = createSelfResolver(config.name)
+  const affinitySources = R.map(R.over(R.lensProp('match'), R.map(resolveSelfToName)))(config.affinity.pod)
```

Comment:
I would do this with just standard JS array methods, or comment the heck of this.  if you don't know ramda this is an entirely alien construct.  i just read up on what `lensProp` and `over` do, but I"m still unsure what this is changing.

Reply:
Commented. Does that clear it up at all?